cli/mender-artifact/copy.go

```diff
@@ -46,15 +46,6 @@ func Cat(c *cli.Context) (err error) {
 	}
 
 	r, err := virtualImage.OpenFile(comp, privateKey, c.Args().First())
-	defer func() {
-		if r == nil {
-			return
-		}
-		cerr := r.Close()
-		if err == nil {
-			err = cerr
-		}
-	}()
```

Comment:
I haven't checked explicitly, but I suspect this will leave all temporary (and very big!) files on the filesystem, so you should check this.

If my suspicion is correct, then a better fix might be to leave this in, but instead have a flag in `r` which indicates whether you want repacking when the handle is closed or not.

Reply:
It will -_-
I was hoping for a quick-fix.
Some handlers already have a `dirty` flag set, when closing. I will extend this to all the other handlers too.